File: src/main/java/nl/hanze/week2/ATM/HasCorrectPin.java
package nl.hanze.week2.ATM;

/**
 * Created by Laurens on 9-10-2017.
 */
public class HasCorrectPin implements ATMState {

    @Override
    public void insertCard(ATM atm) {
        System.out.println("Already present");
    }

    @Override
    public void ejectCard(ATM atm) {
        System.out.println("ejecting card");
        atm.setState(new IdleState());
    }

    @Override
    public void insertPin(ATM atm) {
        System.out.println("Pin already entered");
    }

    @Override
    public void requestAmount(ATM atm) {
        System.out.println("SuccesFully requested amount");
        atm.setState(new HasAmount());
    }
}

File: src/main/java/nl/hanze/week2/assignment45/HasCorrectPin.kt
package nl.hanze.week2.assignment45

/**
 * Created by Laurens on 9-10-2017.
 */
class HasCorrectPin : ATMState {

    override fun insertCard(atm: ATM) = println("Already present")

    override fun ejectCard(atm: ATM) {
        println("ejecting card")
        atm.state = IdleState()
    }

    override fun insertPin(atm: ATM) = println("Pin already entered")

    override fun requestAmount(atm: ATM) {
        println("SuccesFully requested amount")
        atm.state =HasAmount()
    }
}
